### DISCUSSION TITLE
How to render a rect in a fixed coordinate?

### DISCUSSION THREAD

**Discussion by ghmendonca** [emilk/egui#2708] [Q&A]
I have a window that has a layout with 2 columns, and in the first one I want to render some rects but it must be in a fixed coordinates, like, it should start at the point (30.0, 30.0) and end at (50.0, 50.0)

I saw some people saying to use the painter, but I couldn't get to work with painter. I don't even know if I should render this inside a `vertical`, `horizontal` or `horizontal_wrapped` or something else.

Would love if someone could give me an example rendering 2 or 3 rects in fixed coordinates!

**Comment by breadfan** [comment]
Usually some engines used for this primitives, e.g. `macroquad`, or `miniquad`. Egui in general won't let you work with coordinates (not talking about painter)

**Comment by pietervandermeer** [comment]
How about the Ui method put() ? 

https://docs.rs/egui/latest/egui/struct.Ui.html#method.put

This does manual layout according to the API documentation.
